src/meta/processors/admin/SnapShot.cpp

```diff
@@ -13,22 +13,38 @@
 
 namespace nebula {
 namespace meta {
-ErrorOr<cpp2::ErrorCode, std::unordered_map<GraphSpaceID, std::vector<cpp2::CheckpointInfo>>>
+ErrorOr<cpp2::ErrorCode,
+        std::unordered_map<
+            GraphSpaceID,
+            std::pair<nebula::cpp2::PartitionBackupInfo, std::vector<cpp2::CheckpointInfo>>>>
 Snapshot::createSnapshot(const std::string& name) {
     auto retSpacesHosts = getSpacesHosts();
     if (!retSpacesHosts.ok()) {
         return cpp2::ErrorCode::E_STORE_FAILURE;
     }
-    std::unordered_map<GraphSpaceID, std::vector<cpp2::CheckpointInfo>> info;
+    std::unordered_map<
+        GraphSpaceID,
+        std::pair<nebula::cpp2::PartitionBackupInfo, std::vector<cpp2::CheckpointInfo>>>
+        info;
     auto spacesHosts = retSpacesHosts.value();
     for (const auto& spaceHosts : spacesHosts) {
         for (const auto& host : spaceHosts.second) {
             auto status = client_->createSnapshot(spaceHosts.first, name, host).get();
             if (!status.ok()) {
                 return cpp2::ErrorCode::E_RPC_FAILURE;
             }
-            info[spaceHosts.first].emplace_back(
-                apache::thrift::FRAGILE, host, status.value());
+            auto infoPair = status.value();
+            auto it = info.find(spaceHosts.first);
+            if (it != info.cend()) {
+                it->second.second.emplace_back(
+                    apache::thrift::FRAGILE, host, std::move(infoPair.first));
```

Comment:
A suggestion, don't use `apache::thrift::FRAGILE`, there will be version compatibility issues in the future.